### PR TITLE
SUB-2413 | add cluster short name to the scope

### PR DESCRIPTION
**Pull request by refaelm92**
Related to https://cyberarmor-io.atlassian.net/browse/SUB-2413

## PR Type:
Enhancement

___
## PR Description:
This PR introduces an enhancement to the AlertScope struct in the usernotificationreporttypes.go file. A new field, 'ClusterShortName', has been added to the AlertScope struct to provide a short name for the cluster. This change will allow for more concise naming conventions within the scope of alerts.

___
## PR Main Files Walkthrough:
`notifications/usernotificationreporttypes.go`: Added a new field 'ClusterShortName' to the AlertScope struct.

___
## User Description:
Signed-off-by: refaelm <refaelm@armosec.io>

### CODE REVIEW

File: notifications/usernotificationreporttypes.go [hunk 1]
```diff
@@ -86,8 +86,9 @@ type AlertConfig struct {
 }
 
 type AlertScope struct {
-	Cluster    string   `json:"cluster,omitempty" bson:"cluster,omitempty"`
-	Namespaces []string `json:"namespaces,omitempty" bson:"namespaces,omitempty"`
+	Cluster          string   `json:"cluster,omitempty" bson:"cluster,omitempty"`
+	ClusterShortName string   `json:"clusterShortName,omitempty" bson:"clusterShortName,omitempty"`
```
Comment (avrahams) [inline notifications/usernotificationreporttypes.go:90]: why?
Cluster short name is not reliable data - it is only for user display

Comment (avrahams) [reply notifications/usernotificationreporttypes.go:90]: If needed than add the cluster guid and fetch the short name from the cluster in config service

File: notifications/alertchannelapimethods.go [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package notifications
+
+func (ac *AlertChannelAPI) GetDomainScope() []AlertScope {
+	scope := make([]AlertScope, 0, len(ac.Scope))
+
+	for _, enrichedScope := range ac.Scope {
+		scope = append(scope, AlertScope{
```
Comment (RinaO1234) [inline notifications/alertchannelapimethods.go:7]: avoid append and allocate the array before

File: notifications/usernotificationreporttypes.go [hunk 3]
```diff
@@ -90,6 +90,11 @@ type AlertScope struct {
 	Namespaces []string `json:"namespaces,omitempty" bson:"namespaces,omitempty"`
 }
 
+type EnrichedScope struct {
+	AlertScope       `json:",inline" bson:",inline"`
```
Comment (RinaO1234) [inline notifications/usernotificationreporttypes.go:94]: do we need also the bson annotation?

Comment (RinaO1234) [inline notifications/usernotificationreporttypes.go:94]: do we need also bson annotation? isn't this enriched response only for api

File: notifications/alertchannelapimethods.go (hunk 2 above)
Comment (refaelm92) [reply notifications/alertchannelapimethods.go:7]: wdym?

**Review by RinaO1234 — APPROVED** [review]

**Pull request merged**